### PR TITLE
render subpage with some values

### LINKED ISSUE #40: Unpack the data and render the subpage card on the left of the design
## Use Case

_Please provide a use case to help us understand your request in context_

## Acceptance Criteria

when the card on the left looks exactly like [the design ](https://www.figma.com/design/kMIJN6BoOkXIhnVUruIHNT/Website-allencell.org?node-id=1198-6040&t=uhhkiKGvnKLoK3yB-0)
## Details

_Please provide any helpful specifications_

### PR DESCRIPTION
**Pull request by meganrm**
Problem
=======
we want to display the cell line data in the subpage as shown [in the design ](https://www.figma.com/design/kMIJN6BoOkXIhnVUruIHNT/Website-allencell.org?node-id=1198-6040&t=2FDkMryTqL7sbPnJ-0)
closes #40 

Solution
========
Unpacked the data and rendered it in a card. This involved a lot of color updates as well. The design was updated with a [new color palette](https://www.figma.com/design/kMIJN6BoOkXIhnVUruIHNT/Website-allencell.org?node-id=2420-11402&t=2FDkMryTqL7sbPnJ-0), so the colors for the whole app are change to be consistent with the naming scheme of the colors. 

with @rugeli @tyler-foster and @interim17 

## Type of change
Please delete options that are not relevant.

* New feature (non-breaking change which adds functionality)

Steps to Verify:
----------------
1. npm start 
2. got to http://localhost:8000/disease-cell-line/AICS-97/
3. should look like the design 

Or:
https://deploy-preview-38--cell-catalog.netlify.app/disease-cell-line/AICS-97/

Screenshots (optional):
-----------------------
Screenshot 2024-08-27 at 3 34 21 PM

Keyfiles (delete if not relevant):
-----------------------
https://github.com/allen-cell-animated/cell-catalog/pull/38/files#diff-82cc1abec44f9670a1da427352210dbe4c718168dd8139139810debcb06864d2

### CODE REVIEW

File: gatsby-config.js [hunk 1]
```diff
@@ -103,6 +103,7 @@ module.exports = {
     ],
     mapping: {
         "MarkdownRemark.frontmatter.gene": `MarkdownRemark.frontmatter.symbol`,
+        "MarkdownRemark.frontmatter.disease": `MarkdownRemark.frontmatter.name`,
```
Comment (meganrm) [inline gatsby-config.js:106]: this is basically how to do a "join" in this system

File: src/components/Layout.tsx [hunk 2]
```diff
@@ -4,78 +4,15 @@ import { ConfigProvider, Layout as AntLayout } from "antd";
...
-                        defaultColor: DARK_BLUE,
-                    },
-                    Card: {
-                        colorBorder: DARK_GRAY,
-                        lineWidth: 1.5,
-                    },
-                },
-            }}
-        >
+        <ConfigProvider theme={theme}>
```
Comment (meganrm) [inline src/components/Layout.tsx:15]: moved this to it's own file so it's easier to find

File: src/style/colors.sass [hunk 3]
```diff
@@ -1,17 +1,20 @@
...
-    --BLACK: #323233;
-    --border-color: var(--DARK_GRAY);
-    --disabled: var(--DARK_GRAY);
+    --RAIN_SHADOW: #CBCBCC;
+    --SERIOUS_GRAY: #323233;
+
+    --ALLEN_LIGHT_10: #DFE5EA;
+    --ALLEN_LIGHT_30: #9FB1C0;
+    --ALLEN_LIGHT_60: #607E96;
+    --ALLEN_BLUE: #003057;
```
Comment (meganrm) [inline src/style/colors.sass:11]: name changes to match design names

File: src/style/disease-catalog.module.css [hunk 4]
```diff
@@ -11,7 +11,7 @@
 .header p {
     font-size: 20px;
     font-weight: 400;
-    color: var(--BLACK); 
+    color: var(--SERIOUS_GRAY);
```
Comment (meganrm) [inline src/style/disease-catalog.module.css:14]: can ignore this file: only changes are to names of color variables

File: src/style/disease-table.module.css
Comment (meganrm) [inline src/style/disease-table.module.css:1]: can ignore this file: only changes are to names of color variables

File: src/style/layout.module.css
Comment (meganrm) [inline src/style/layout.module.css:1]: can ignore this file: only changes are to names of color variables

File: src/style/modal.module.css
Comment (meganrm) [inline src/style/modal.module.css:1]: can ignore this file: only changes are to names of color variables

File: src/style/theme.ts [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+        Descriptions: {
+            itemPaddingBottom: 0,
+        },
+        Tag: {
+            defaultColor: primaryColor,
+        },
+        Card: {
+            colorBorder: RAIN_SHADOW,
+            lineWidth: 1.5,
+        },
```
Comment (meganrm) [inline src/style/theme.ts:66]: only addition is this card field, everything else just moved

File: src/templates/disease-cell-line.tsx [hunk 6]
```diff
@@ -1,60 +1,218 @@
-import React from "react";
-import PropTypes from "prop-types";
-import { graphql } from "gatsby";
+import React, { useState } from "react";
```
Comment (meganrm) [inline src/templates/disease-cell-line.tsx:1]: this is the file that renders the subpage

**Comment by meganrm** [comment]
There is a unique "key" warning that I can't run down. Any help would be appreciated

**Comment by interim17** [comment]
> There is a unique "key" warning that I can't run down. Any help would be appreciated

The error is coming from the clones array being passed to the `dataSource` prop in `CloneTable.tsx`

Can remove the error by providing an arbitrary key before passing in the prop:
```
  const dataSourceWithKey = dataSource.map((data, index) => {
    return { ...data, key: index };
  });
```
Currently most of the data fields for clones are optional but it seems like "clone number" will be unique and could maybe be used as key in the future?

File: src/templates/disease-cell-line.tsx [hunk 7]
```diff
@@ -1,60 +1,221 @@
...
+interface DiseaseCellLineTemplateProps {
+    href: string;
+    cellLineId: string;
+    geneName: string;
+    geneSymbol: string;
+    status: string;
+    snp: string;
+    orderLink: string;
+    certificateOfAnalysis: string;
+    parentalLine: any;
```
Comment (ShrimpCryptid) [inline src/templates/disease-cell-line.tsx:38]: Why is `parentalLine` `any` here?

File: src/templates/disease-cell-line.tsx [hunk 8]
```diff
@@ -1,60 +1,221 @@
...
+    geneSymbol,
+    clones,
+    snp,
+    orderLink,
+    certificateOfAnalysis,
+    parentalLine,
+    healthCertificate,
+}: DiseaseCellLineTemplateProps) => {
+    const defaultToolTipText = "Copy cell line link to clipboard";
+    const [toolTipText, setToolTipText] = useState(defaultToolTipText);
```
Comment (ShrimpCryptid) [inline src/templates/disease-cell-line.tsx:58]: Nit: Name to indicate that this tooltip is only for the share button?

File: src/templates/disease-cell-line.tsx [hunk 9]
```diff
@@ -1,60 +1,221 @@
...
+        },
+        {
+            key: "4",
+            label: "Parental Line",
+            children: `${formatCellLineId(parentalLine.cell_line_id)} cl. ${
+                parentalLine.clone_number
+            } ${parentLineGene.symbol}`,
+        },
+    ];
+    const urlToCopy = `Check out this cell line from the Allen Cell Collection! ${href}`;
```
Comment (ShrimpCryptid) [inline src/templates/disease-cell-line.tsx:83]: I feel like I usually only see this pattern when sharing is done on mobile devices via the native sharing API between apps... as a user I'd expect to only get the URL, and not the additional text. Worth discussing with UX if this didn't come directly from them?

File: src/templates/disease-cell-line.tsx [hunk 10]
```diff
@@ -1,60 +1,221 @@
...
+                        </DefaultButton>
+                    </Flex>
+                    <Button
+                        type="primary"
+                        className={extraLargeButton}
+                        href={orderLink}
+                        target="_blank"
+                        rel="noreferrer"
+                    >
+                        <h2>Obtain {formatCellLineId(cellLineId)}</h2>
```
Comment (ShrimpCryptid) [inline src/templates/disease-cell-line.tsx:157]: Missing external link icon from designs?

File: src/templates/disease-cell-line.tsx (hunk 7 above)
Comment (meganrm) [reply src/templates/disease-cell-line.tsx]: it was a placeholder that I forgot to go back to, thanks for the catch !

**Comment by meganrm** [comment]
> > There is a unique "key" warning that I can't run down. Any help would be appreciated
> 
> @meganrm Seems the error is coming from the clones array being passed to the `dataSource` prop in `CloneTable.tsx`
> 
> Can remove the error by providing an arbitrary key before passing in the prop:
> 
> ```
>   const dataSourceWithKey = dataSource.map((data, index) => {
>     return { ...data, key: index };
>   });
> ```
> 
> Currently most of the data fields for clones are optional but it seems like "clone number" will be unique and could maybe be used as key in the future?

Thank you!!

**Review by ShrimpCryptid — APPROVED** [review]
LGTM!

**Pull request merged**